exercises/nodejs/src/game.js

```diff
@@ -1,34 +1,33 @@
 import generator from 'random-seed'
 
 var Game = function () {
-  var players = new Array()
-  var places = new Array(6)
-  var purses = new Array(6)
-  var inPenaltyBox = new Array(6)
+  const players = new Array()
+  const places = new Array(6)
+  const purses = new Array(6)
+  const inPenaltyBox = new Array(6)
 
-  var popQuestions = new Array()
-  var scienceQuestions = new Array()
-  var sportsQuestions = new Array()
-  var rockQuestions = new Array()
+  const popQuestions = new Array()
+  const scienceQuestions = new Array()
+  const sportsQuestions = new Array()
+  const rockQuestions = new Array()
 
-  var currentPlayer = 0
-  var isGettingOutOfPenaltyBox = false
+  const numberOfCategories = 4
+  let currentPlayer = 0
+  let isGettingOutOfPenaltyBox = false
 
   var didPlayerWin = function () {
     return !(purses[currentPlayer] == 6)
   }
 
   var currentCategory = function () {
-    if (places[currentPlayer] == 0) { return 'Pop' }
-    if (places[currentPlayer] == 4) { return 'Pop' }
-    if (places[currentPlayer] == 8) { return 'Pop' }
-    if (places[currentPlayer] == 1) { return 'Science' }
-    if (places[currentPlayer] == 5) { return 'Science' }
-    if (places[currentPlayer] == 9) { return 'Science' }
-    if (places[currentPlayer] == 2) { return 'Sports' }
-    if (places[currentPlayer] == 6) { return 'Sports' }
-    if (places[currentPlayer] == 10) { return 'Sports' }
-    return 'Rock'
+    const categoryNumber = places[currentPlayer] % numberOfCategories
+
+    switch (categoryNumber) {
+      case 0: return 'Pop'
+      case 1: return 'Science'
+      case 2: return 'Sports'
+      case 3: return 'Rock'
+    }
```

Comment:
How do you feel about switch statements? I find them to be a bit unintuitive. The returns in yours mask the fact that usually these have `break`s. I like the use of modulus, but perhaps this could be used in four conditionals and hidden behind a query function? See [Replace Temp with Query](https://refactoring.guru/replace-temp-with-query).